src/librustc_mir/build/matches/simplify.rs

```diff
@@ -195,6 +195,10 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                 candidate.match_pairs.push(MatchPair::new(place, subpattern));
                 Ok(())
             }
+
+            PatternKind::Or { .. } => {
+                Err(match_pair)
```

Comment:
Is it possible for the vector to be length 1 here?

Reply:
I think the pattern `Some(0 |)` will have length 1, in which case we can return `Ok(())` here (or normalise it earlier on).

Reply:
We currently don't parse a trailing `|` as a valid pattern, so the vector would be at least length 2 right?

Reply:
We currently don't parse a trailing `|` as a valid pattern, so the vector would be at least length 2 right?

Reply:
Sorry, yes, that's true. Can you add a comment above `Err` saying that the vector will always have length at least 2, and so will always be refutable?